specification/automation/resource-manager/Microsoft.Automation/stable/2018-01-15/dscNode.json

```diff
@@ -0,0 +1,957 @@
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./definitions.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./definitions.json#/parameters/AutomationAccountNameParameter"
+          },
+          {
+            "name": "nodeId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The Dsc node id."
+          },
+          {
+            "name": "reportId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The report id."
+          },
+          {
+            "$ref": "./definitions.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "./definitions.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/DscNodeReport"
+            }
+          },
+          "default": {
+            "description": "Automation error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./definitions.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Automation/automationAccounts/{automationAccountName}/nodes/{nodeId}/reports/{reportId}/content": {
+      "get": {
+        "tags": [
+          "NodeReports"
+        ],
+        "operationId": "NodeReports_GetContent",
+        "description": "Retrieve the Dsc node reports by node id and report id.",
+        "externalDocs": {
+          "url": "http://aka.ms/azureautomationsdk/dscnodereportoperations"
+        },
+        "x-ms-examples": {
+          "List DSC reports by node id.": {
+            "$ref": "./examples/listDscNodeReportsByNode.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./definitions.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./definitions.json#/parameters/AutomationAccountNameParameter"
+          },
+          {
+            "name": "nodeId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The Dsc node id."
+          },
+          {
+            "name": "reportId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The report id."
+          },
+          {
+            "$ref": "./definitions.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "./definitions.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "type": "file"
```

Comment:
>file [](start = 23, length = 4)

This is not what I see in the report. It looks like this could be defined as a model

Reply:
I think the example is wrong in this case. It does not return a well-formed JSON

It sends something like this: 
URL: `https://management.azure.com/subscriptions/b5e4748c-f69a-467c-8749-e2f9c8cd3db0/resourceGroups/ConfigDataValidation/providers/Microsoft.Automation/automationAccounts/ConfigDataValidation/nodes/119a0b96-23fd-11e8-b29c-000d3a18dec4/reports/76f09587-2c84-11e8-a9ca-000d3a18dec4/content?api-version=2018-01-15`

The Response - which is raw reports from our DSC Nodes.
```
{"JobId":"76f09587-2c84-11e8-a9ca-000d3a18dec4","OperationType":"Consistency","NodeName":"ANAGG-2008R2","IpAddress":"10.13.49.8;127.0.0.1;fe80::2cc0:8062:a210:e1c6%11;::2000:0:0:0;::1;::2000:0:0:0","LCMVersion":"2.0","ReportFormatVersion":"2.0","StartTime":"2018-03-20T21:20:09.0080000+00:00","Errors":[],"StatusData":["{\"IPV4Addresses\":[\"10.13.49.8\",\"127.0.0.1\"],\"MACAddresses\":[\"00-0D-3A-18-DE-C4\",\"00-00-00-00-00-00-00-E0\"],\"Type\":\"Consistency\",\"HostName\":\"ANAGG-2008R2\",\"Locale\":\"en-US\",\"StartDate\":\"2018-03-20T21:20:09.0080000+00:00\",\"JobID\":\"{76F09587-2C84-11E8-A9CA-000D3A18DEC4}\",\"LCMVersion\":\"2.0\",\"IPV6Addresses\":[\"fe80::2cc0:8062:a210:e1c6%11\",\"::2000:0:0:0\",\"::1\",\"::2000:0:0:0\"]}"],"AdditionalData":[{"Key":"OSVersion","Value":"{\"VersionString\":\"Microsoft Windows NT 6.1.7601 Service Pack 1\",\"ServicePack\":\"Service Pack 1\",\"Platform\":\"Win32NT\"}"},{"Key":"PSVersion","Value":"{\"CLRVersion\":\"4.0.30319.42000\",\"PSVersion\":\"5.1.14409.1012\",\"BuildVersion\":\"10.0.14409.1012\"}"}]}

{"JobId":"76f09587-2c84-11e8-a9ca-000d3a18dec4","OperationType":"Consistency","RefreshMode":"Pull","Status":"Success","ReportFormatVersion":"2.0","StartTime":"2018-03-20T21:20:09.0080000+00:00","EndTime":"2018-03-20T21:20:37.0080000+00:00","RebootRequested":"False","Errors":[],"StatusData":["{\"StartDate\":\"2018-03-20T21:20:09.0080000+00:00\",\"IPV6Addresses\":[\"fe80::2cc0:8062:a210:e1c6%11\",\"::2000:0:0:0\",\"::1\",\"::2000:0:0:0\"],\"MACAddresses\":[\"00-0D-3A-18-DE-C4\",\"00-00-00-00-00-00-00-E0\"],\"DurationInSeconds\":\"28\",\"Status\":\"Success\",\"IPV4Addresses\":[\"10.13.49.8\",\"127.0.0.1\"],\"LCMVersion\":\"2.0\",\"Type\":\"Consistency\",\"HostName\":\"ANAGG-2008R2\",\"RebootRequested\":\"False\",\"JobID\":\"{76F09587-2C84-11E8-A9CA-000D3A18DEC4}\",\"MetaConfiguration\":{\"AgentId\":\"119A0B96-23FD-11E8-B29C-000D3A18DEC4\",\"SignatureValidations\":[],\"ConfigurationDownloadManagers\":[{\"RegistrationKey\":\"\",\"ServerURL\":\"https://eus2-agentservice-prod-1.azure-automation.net/accounts/bd8fac9e-01c5-4523-98e0-f273f474fbf6\",\"ResourceId\":\"[ConfigurationRepositoryWeb]AzureAutomationDSC\",\"ConfigurationNames\":[],\"SourceInfo\":\"C:\\\\Packages\\\\Plugins\\\\Microsoft.Powershell.DSC\\\\2.75.0.0\\\\DSCWork\\\\RegistrationMetaConfigV2.0\\\\RegistrationMetaConfigV2.ps1::69::9::ConfigurationRepositoryWeb\"}],\"ActionAfterReboot\":\"ContinueConfiguration\",\"LCMCompatibleVersions\":[\"1.0\",\"2.0\"],\"LCMState\":\"Idle\",\"ResourceModuleManagers\":[{\"RegistrationKey\":\"\",\"ServerURL\":\"https://eus2-agentservice-prod-1.azure-automation.net/accounts/bd8fac9e-01c5-4523-98e0-f273f474fbf6\",\"ResourceId\":\"[ResourceRepositoryWeb]AzureAutomationDSC\",\"SourceInfo\":\"C:\\\\Packages\\\\Plugins\\\\Microsoft.Powershell.DSC\\\\2.75.0.0\\\\DSCWork\\\\RegistrationMetaConfigV2.0\\\\RegistrationMetaConfigV2.ps1::76::9::ResourceRepositoryWeb\"}],\"ReportManagers\":[{\"RegistrationKey\":\"\",\"ServerURL\":\"https://eus2-agentservice-prod-1.azure-automation.net/accounts/bd8fac9e-01c5-4523-98e0-f273f474fbf6\",\"ResourceId\":\"[ReportServerWeb]AzureAutomationDSC\",\"SourceInfo\":\"C:\\\\Packages\\\\Plugins\\\\Microsoft.Powershell.DSC\\\\2.75.0.0\\\\DSCWork\\\\RegistrationMetaConfigV2.0\\\\RegistrationMetaConfigV2.ps1::82::9::ReportServerWeb\"}],\"StatusRetentionTimeInDays\":\"10\",\"LCMVersion\":\"2.0\",\"MaximumDownloadSizeMB\":\"500\",\"ConfigurationMode\":\"ApplyAndMonitor\",\"RefreshFrequencyMins\":\"30\",\"RebootNodeIfNeeded\":\"False\",\"SignatureValidationPolicy\":\"NONE\",\"RefreshMode\":\"Pull\",\"DebugMode\":[\"NONE\"],\"CertificateID\":\"24C07ACB944D2781BDB53E2CB636CEE890BDF39E\",\"AllowModuleOverwrite\":\"False\",\"ConfigurationModeFrequencyMins\":\"15\",\"LCMStateDetail\":\"\"},\"Locale\":\"en-US\",\"Mode\":\"Pull\"}"],"AdditionalData":[{"Key":"OSVersion","Value":"{\"VersionString\":\"Microsoft Windows NT 6.1.7601 Service Pack 1\",\"ServicePack\":\"Service Pack 1\",\"Platform\":\"Win32NT\"}"},{"Key":"PSVersion","Value":"{\"CLRVersion\":\"4.0.30319.42000\",\"PSVersion\":\"5.1.14409.1012\",\"BuildVersion\":\"10.0.14409.1012\"}"}]}

```

Reply:
Still looks like something that can be modeled

Reply:
We don't want to model this currently. As this is subject to change based on what the agent sends and we don't want this to be a breaking change. Please let me know if this is a blocker for this PR.

Reply:
What about returning a model with `additionalProperties` ?